Document/0x06j-Testing-Resiliency-Against-Reverse-Engineering.md

```diff
@@ -523,6 +523,93 @@ A similar approach works. Answer the following questions:
+
+#### Effectiveness Assessment
+
+Launch the app with various reverse engineering tools and frameworks installed in your test device. Include at least the following: Frida, Cydia Substrate, Cycript, SSL KillSwitch and the Needle Agent.
+
+The app should respond in some way to the presence of those tools. For example by:
+
+- Alerting the user and asking for accepting liability.
+- Preventing execution by gracefully terminating.
+- Securely wiping any sensitive data stored on the device.
+- Reporting to a backend server, e.g, for fraud detection.
+
+Next, work on bypassing the detection of the reverse engineering tools and answer the following questions:
+
+- Can the mechanisms be bypassed trivially (e.g., by hooking a single API function)?
+- How difficult is identifying the anti reverse engineering code via static and dynamic analysis?
+- Did you need to write custom code to disable the defenses? How much time did you need?
+- What is your assessment of the difficulty of bypassing the mechanisms?
+
+The following steps should guide you when bypassing detection of reverse engineering tools:
+
+1. Patch the anti reverse engineering functionality. Disable the unwanted behavior by patching the binary through usage of Hooper or Ghidra.
+2. Use Frida or Cydia Substrate to hook file system APIs on the Objective-C /  Swift or native layers. Return a handle to the original file, not the modified file.
+
+Refer to the chapter [Tampering and Reverse Engineering on iOS](0x06c-Reverse-Engineering-and-Tampering.md)" for examples of patching and code injection.
+
+### Testing Emulator Detection (MSTG-RESILIENCE-5)
+
+#### Overview
+
+In the context of anti-reversing, the goal of emulator detection is to increase the difficulty of running the app on an emulated device, which impedes some tools and techniques reverse engineers like to use. This increased difficulty forces the reverse engineer to defeat the emulator checks or utilize the physical device, thereby barring the access required for large-scale device analysis.
+
+To make a long story short, this is not applicable for iOS apps. The only available simulator is the one that ships with Xcode. Simulator binaries are compiled to x86 code instead of ARM code and apps compiled for a real device (ARM architecture) don't run in the simulator. This makes the simulator useless for black box analysis and reverse engineering. See also the section [Testing on the iOS Simulator](0x06b-Basic-Security-Testing.md "Testing on the iOS Simulator") in the basic security testing chapter for further information.
```

Comment:
```suggestion
This is however not a concern on iOS. The only available simulator is the one that ships with Xcode. Simulator binaries are compiled to x86 code instead of ARM code and apps compiled for a real device (ARM architecture) don't run in the simulator. This makes the simulator useless for black box analysis and reverse engineering. See also the section [Testing on the iOS Simulator](0x06b-Basic-Security-Testing.md "Testing on the iOS Simulator") in the basic security testing chapter for further information.
```
(We can only make a long story short, if we have the long story)

Reply:
fair enough ;-)